lib/utils/error-messages.js

```diff
@@ -339,72 +343,57 @@ function getPropertyRequiredErrorMessage(report, executionPlatformLabel) {
     return `${ getIndefiniteArticle(typeString) } <${ typeString }> must have a defined <Timer duration>`;
   }
 
-  if (is(node, 'bpmn:FormalExpression')
-    && requiredProperty === 'body'
-    && TIMER_PROPERTIES.includes(secondLast(report.path))
-  ) {
-    return `${ getIndefiniteArticle(typeString) } <${ typeString }> must have a defined <Timer value>`;
-  }
-
   return message;
 }
 
-function getPropertyTypeNotAllowedErrorMessage(report, executionPlatformLabel) {
+function getExpressionRequiredErrorMessage(report) {
   const {
-    error,
-    message
+    data
   } = report;
 
   const {
     node,
     parentNode,
     property
-  } = error;
+  } = data;
 
   const typeString = getTypeString(parentNode || node);
 
-  if (is(node, 'bpmn:Event') && property === 'eventDefinitions') {
-    return `${ getIndefiniteArticle(typeString) } <${ typeString }> is not supported by ${ executionPlatformLabel }`;
+  if (is(node, 'bpmn:FormalExpression') && TIMER_PROPERTIES.includes(property)) {
+    return `${ getIndefiniteArticle(typeString) } <${ typeString }> must have a defined <Timer value>`;
   }
-
-  return message;
 }
 
-function getPropertyValueNotAllowedErrorMessage(report, executionPlatformLabel) {
+function getExpressionValueNotAllowedErrorMessage(report) {
   const {
-    error
+    data
   } = report;
 
   const {
     node,
     parentNode,
     property
-  } = error;
+  } = data;
 
   const typeString = getTypeString(parentNode || node);
 
-  if (is(node, 'bpmn:FormalExpression')
-    && property === 'body'
-    && secondLast(report.path) === 'timeCycle'
-  ) {
+  if (is(node, 'bpmn:FormalExpression') && property === 'timeCycle') {
     return `${ getIndefiniteArticle(typeString) } <${ typeString }> <Time cycle> should be an expression, an ISO 8601 repeating interval, or a cron expression (cron requires Camunda Platform 8.1 or newer)`;
   }
 
-  if (is(node, 'bpmn:FormalExpression')
-    && property === 'body'
-    && secondLast(report.path) === 'timeDate'
-  ) {
+  if (is(node, 'bpmn:FormalExpression') && property === 'timeDate') {
     return `${ getIndefiniteArticle(typeString) } <${ typeString }> <Time date> should be an expression, or an ISO 8601 date`;
   }
 
-  if (is(node, 'bpmn:FormalExpression')
-    && property === 'body'
-    && secondLast(report.path) === 'timeDuration'
-  ) {
+  if (is(node, 'bpmn:FormalExpression') && property === 'timeDuration') {
     return `${ getIndefiniteArticle(typeString) } <${ typeString }> <Time duration> should be an expression, or an ISO 8601 interval`;
   }
 }
 
-function secondLast(array) {
-  return array[array.length - 2];
-}
+function getSupportedMessage(prefix, executionPlatformVersion, allowedVersion) {
+  if (allowedVersion) {
+    return `${ prefix } is only supported by ${ getExecutionPlatformLabel('Camunda Cloud', allowedVersion) } or newer`;
+  }
+
+  return `${ prefix } is not supported by ${ getExecutionPlatformLabel('Camunda Cloud', executionPlatformVersion) }`;
+}
```

Comment:
We are hard-coding the Execution platform to `Camunda Cloud` here. This assumes that this error type/utility will never be used for Camunda 7. 
Is there a reason we are not using the platform from the XML?https://github.dev/camunda/linting/blob/8a26cbef0231bb16e83122a3044323ed13c323c9/lib/Linter.js#L41-L43

Reply:
Right now there is no use case for Camunda 7. But you're right, we could use the execution platform from the XML. So, in that case, we'd also assume that `allowedVersion` refers to the same execution platform. I'll adjust.

Reply:
Done: https://github.com/camunda/linting/pull/34/commits/afac378d99e232ad48bf9da59ba238da3a8f2a7c ✔️